File: rb/features.rb
# CountriesAndCities SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/ratelimit_feature'
require_relative 'feature/retry_feature'
require_relative 'feature/test_feature'
require_relative 'feature/timeout_feature'


module CountriesAndCitiesFeatures
  def self.make_feature(name)
    case name
    when "base"
      CountriesAndCitiesBaseFeature.new
    when "ratelimit"
      CountriesAndCitiesRatelimitFeature.new
    when "retry"
      CountriesAndCitiesRetryFeature.new
    when "test"
      CountriesAndCitiesTestFeature.new
    when "timeout"
      CountriesAndCitiesTimeoutFeature.new
    else
      CountriesAndCitiesBaseFeature.new
    end
  end
end

File: py/features.py
# CountriesAndCities SDK feature factory

from feature.base_feature import CountriesAndCitiesBaseFeature
from feature.test_feature import CountriesAndCitiesTestFeature


def _make_feature(name):
    features = {
        "base": lambda: CountriesAndCitiesBaseFeature(),
        "test": lambda: CountriesAndCitiesTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
